src/ccache.cpp

```diff
@@ -1559,15 +1559,15 @@ get_result_name_from_cpp(struct args* args, struct hash* hash)
 
     // Limit the basename to 10 characters in order to cope with filesystem with
     // small maximum filename length limits.
-    string_view input_base = Util::base_name(input_file);
-    size_t dot_pos = input_base.find('.');
-    size_t truncate_pos = std::min(size_t(10),
-                                   std::min(input_base.size(), dot_pos));
-    input_base = input_base.substr(0, truncate_pos);
-
-    path_stdout = x_strdup(
-        fmt::format("{}/{}.stdout", temp_dir(), input_base).c_str());
-
+    path_stdout = []() -> char* {
+      string_view input_base = Util::base_name(input_file);
+      size_t dot_pos = input_base.find('.');
+      size_t truncate_pos = std::min(size_t(10),
+                                     std::min(input_base.size(), dot_pos));
+      input_base.substr(0, truncate_pos);
+      return x_strdup(
+          fmt::format("{}/{}.stdout", temp_dir(), input_base).c_str());
+    }();
```

Comment:
Ok, the "improvement" looks terrible with this diff. Mostly to visually separate the logic and the variable noise it adds. But completely optional.

Reply:
How about extracting the truncation logic to a named function (`get_truncated_base_name`?)?Then we could unit test it as well.